configs/plugins/mermaid-reporter-plugin.js

```diff
@@ -0,0 +1,109 @@
+/* eslint-disable security/detect-object-injection */
+const mermaidNode = (pNode, pText) => {
+  const lNode = pNode
+    .replace(/^\.$|^\.\//g, "__currentPath__")
+    .replace(/^\.{2}$|^\.{2}\//g, "__prevPath__")
+    .replace(/[[\]/.@]/g, "");
```

Comment:
The syntax of mermaid.js seems to require escaping of meta characters in the node.

```
// OK
foo[foo] ---> bar[bar]

// NG
./foo[./foo] ---> ../bar[../bar]
```
Text can be escaped by double quotes, but node cannot, so the approach of removing meta characters seems to be the way to go.

```
// OK
__currentPath__/foo["./foo"] ---> __prevPath__/bar["../bar"]
```

Reply:
A good find. 
With the last replace it seems we'll be losing information potentially creating collisions. E.g.

src/an/other/index.js => srcanotherindexjs
src/another/index.js => srcanotherindexjs

this might be solved by using a replacement character instead? e.g. 

```suggestion
    .replace(/[[\]/.@]/g, "_");
```

src/an/other/index.js => src_an_otherindex_js
src/another/index.js => src_another_index_js

Reply:
That sounds better! No collisions, no syntax issues, and high visibility for humans!
I'll revise this policy.